### PR TITLE
feat: Testing. More ways to verify table content.

### PR DESCRIPTION
**Pull request by daniil-ushkov**
Hi! 
In [yandex-cloud provider](https://github.com/yandex-cloud/cq-provider-yandex) we want to verify more complex conditions than just non-empty columns. So I suggest to provide opportunity of writing custom verifiers for user.

### CODE REVIEW

File: provider/testing/verifiers.go [hunk 1]
```diff
@@ -0,0 +1,80 @@
+package testing
```
Comment (roneli) [inline provider/testing/verifiers.go:1]: Would add documentation for each verifier type, and what they are checking

File: provider/testing/resource.go [hunk 2]
```diff
@@ -32,8 +32,11 @@ type ResourceTestCase struct {
 	ParallelFetchingLimit uint64
 	// SkipIgnoreInTest flag which detects if schema.Table or schema.Column should be ignored
 	SkipIgnoreInTest bool
+	Verifiers        []Verifier
 }
 
+type Verifier func(t *testing.T, conn pgxscan.Querier)
```
Comment (roneli) [inline provider/testing/resource.go:38]: would maybe make the verifier signature like this, and the default verifier will be no empty columns
```suggestion
type Verifier func(t *testing.T, table *schema.Table, conn pgxscan.Querier, SkipIgnoreInTest bool)
```

File: provider/testing/resource.go [hunk 3]
```diff
@@ -32,8 +32,11 @@ type ResourceTestCase struct {
 	ParallelFetchingLimit uint64
 	// SkipIgnoreInTest flag which detects if schema.Table or schema.Column should be ignored
 	SkipIgnoreInTest bool
+	Verifiers        []Verifier
```
Comment (roneli) [inline provider/testing/resource.go:35]: nit: document this field

**Review by roneli — CHANGES_REQUESTED** [review]
Looks like a great addition to SDK testing, had a few suggestions

File: provider/testing/resource.go (hunk 2 above)
Comment (daniil-ushkov) [reply provider/testing/resource.go]: To support ignoring flag, I changed signature as you suggested. Now single verifier will be used to verify one or more tables (main table and relations) from specific *schema.Table. So in ResourceTestCase's Verifiers field we have map from resource name to list of verifiers for this resource.

**Review by roneli — APPROVED** [review]
LGTM

**Comment by yevgenypats** [comment]
Hi @daniil-ushkov Looks great, thx! I'd love to run a few more potential improvement that we are thinking on the SDK side. Can you drop me a note on yp@cloudquery.io (couldn't find your email easily :) )?

**Comment by shimonp21** [comment]
@daniil-ushkov Hi :) Updating that we released version [v0.8.19](https://github.com/cloudquery/cq-provider-sdk/releases/tag/v0.8.19) of the SDK.

**Pull request merged**